### PR TITLE
Low level components: Meatballs, tabs, buttons, icons, and deks

### PR DESCRIPTION
**Pull request by aaronalbright**
## Components

* `<BaseIcon/>`
  * Used for generating a HeroIcon.
  * `iconName`: Name of icon as it exists in the library. Will always end in `Icon`. (e.g. `"ArchiveBoxIcon"`)
  * `size`: One of two sizes to use. Defaults to 20px version. Optionally pass `size="24"` to the more-detailed icon suitable for larger sizes like for us in `<IconButton>`
* `<Button/>`
  * Basic button. Only things of the note are the optional `value` prop if you don't want to deal with children.
  * Has an `icon` parameter for passing in an icon componant. Can either be instance of `<BaseIcon>` or a component from `@heroicons`. This is different from passing an `iconName` because a) I made this component first and b) There's only two places where we ever have an icon in one of these buttons anyway.
* `<IconButton/>`
  * Similar to Button.
  * Utilizes `<BaseIcon>`'s `iconName` prop to determine the icon shown alongside the label.
  * `setWhite` prop is used when the button is on the CMS Prview toolbar with unique styilng for that visual context
* `<MeatballItem/>`
  * Same use of `iconName` for icon.
  * Optional `danger` boolean for delete actions.
  * `action` prop to trigger an action on click. I suppose this could just be `onClick` for consistency with the other actual buttons.
* `<ProjectStatusDek/>`
  * Accepts `status` prop of `published` `archive` or `uploaded`. "Default" is considered to be in a local, unuplaoded state with the prompt to backup via `<BackupPrompt>`
  * Uses relative time for anything less than 24 hours. (e.g. "Last backuped up: 15 minutes ago" "Last published: 2 hours ago") I can tighen up that to within 12 hours or less if needed. Otherwise display full date time as planned "Oct 24, 2022 2:23pm".
* `<ProjectStatusIcon/>`
  * Simple status icon that shares `status` state of `published` or `archive`. Any other state, uploaded or not display the circle with the dash. Custom icon was made for the Archive status because dealing with CSS was more annoying that just cutting a new SVG.
* `<TabToggleItem/>`
  * NOTE: `Tab.Group` is a mess since I didn't want to fully commit to that parent component until this was done. But I worked ahead more than I expected. Luckily, it'll make less work to do when I do end up making the proper `TabToggle` component.
  * Accepts `size` property for icon style/size (The 24px device icons look better than the 20px ones)
  * There's no state management or "active" tab managed because the `<Tab.Group>` component handles that at the parent via `selectedIndex` property. Though, `<Tab>` does expose a `selected` render prop, but it's mostly used for styling. Which we don't need since that also handled at the parent level for the animation.
* `<EmptyProject>`
  * Not much to it. Shows what an empty project would look like with super basic scaffolding of what the actual functional components will look like.

## How to demo

Inside app view there is an `isEmpty` variable. Set to `true` to see the `<EmptyProject>` component. Otherwise you'll just see all the other components laid out in demo form with occasional style context as needed.

components

### CODE REVIEW

File: src/components/Button/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.btn {
+  border-radius: 9999px;
```
Comment (brizandrew) [inline src/components/Button/styles.module.css:2]: I think we have a theme for this now right?

File: src/components/Button/styles.module.css [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  color: var(--slate-100);
+  background-color: var(--slate-800);
+}
+
+.outline:active {
+  filter: brightness(1.33);
+}
+
+.ghost {
+  color: #334155;
```
Comment (brizandrew) [inline src/components/Button/styles.module.css:39]: In elections we kept all colors together as variables. I think it's worth keeping that paradigm around, it made simplifying colors much easier.

File: src/components/IconButton/index.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    className = '',
+    onClick = () => null,
+    iconName,
+    label,
+    setWhite = false,
+  } = props;
+
+  const btnClass = cls(
+    styles.btn,
+    setWhite && styles.white,
```
Comment (brizandrew) [inline src/components/IconButton/index.jsx:33]: So classnames actually let's you do something like

```
{ [styles.white]: setWhite } 
```

which definitely looks silly as one of a number of list, but generally I just make the last argument the "optional" classes argument which is an object like that above. Makes it easier to quicly read, what are styles that are always applied, and which are conditional.

File: src/components/MeatballItem/index.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          size="20"
+          iconName={iconName}
+          className={cls(styles.iconSm, margin.mr2)}
+        />
+        {label}
+      </button>
+    );
+  };
+
+  return <Menu.Item>{renderItem}</Menu.Item>;
```
Comment (brizandrew) [inline src/components/MeatballItem/index.jsx:56]: So here, `renderItem` is a function that returns a node. To me it feels like it should actually be a component then which you import and then use something like

```
return (
  <Menu.Item>
    {MenuItem}
  </Menu.Item>
)
```

Any glaring reason not to do this?

File: src/components/ProjectStatusIcon/index.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    if (status === 'published') {
+      return <CheckCircleIcon />;
+    }
+    if (status === 'archive') {
+      return (
+        <ArchiveIcon />
+      );
+    }
+    return <MinusCircleIcon />;
+  };
```
Comment (brizandrew) [inline src/components/ProjectStatusIcon/index.jsx:30]: So in keeping with our only one return node per component, this should proabably be a map you set up outside this component

```
const ICONS = {
  published: CheckCirlceIcon,
  archive: ArchiveIcon,
  ...
}
```

Then in the component you can

```
const IconComponent = ICONS[status] || DefaultIcon
```

**Review by brizandrew — COMMENTED** [review]
Most of this looks really great! A few tweaks here and there, but mostly keep this going for the rest! Thanks for all the great work!

File: src/components/MeatballItem/index.jsx (hunk 4 above)
Comment (aaronalbright) [reply src/components/MeatballItem/index.jsx:56]: Couldn't figure out a way to do it like that while also using the `active` render prop from `<Menu.Item>` without getting some weird ref-forwarding related errors. I refactored a bit to have all of the node rendering occur in main return with a simple function for class name logic.

**Pull request merged**